{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.chdir('/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/process_alignments')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import Bio\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio import Entrez\n",
    "from time import process_time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### First step) Reading the sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">SUP35_Kla_AB039749\n",
      "ATGTCAGACCAACAAAATCAAGACCAAGGGCAAGGCCAAGGTTACAATCAGTATAACCAATATGGCCAGT\n",
      "ACAACCAGTACTACAACCAACAGGGCTATCAAGGCTACAACGGCCAACAAGGTGCTCCTCAAGGCTACCA\n",
      "AGCATATCAAGCTTATGGACAGCAGCCTCAAGGAGCCTACCAGGGCTACAACCCTCAACAAGCTCAAGGC\n",
      "TACCAACCTTACCAGGGCTACAATGCTCAGCAACAAGGTTACAACGCTCAGCAAGGCGGTCACAACAATA\n",
      "ACTACAATAAAAATTATAATAATAAAAACAGTTACAATAACTATAATAAGCAGGGTTATCAAGGTGCTCA\n",
      "AGGATATAATGCACAACAGCCAACCGGTTACGCTGCTCCAGCACAGTCTTCATCCCAGGGTATGACTTTG\n",
      "AAAGATTTCCAAAACCAACAAGGCAGTACTAATGCAGCCAAGCCAAAGCCTAAGTTAAAGTTGGCCTCTA\n",
      "GCTCTGGTATTAAGTTAGTAGGTGCCAAGAAACCTGTAGCACCCAAAACTGAGAAAACTGATGAATCCAA\n",
      "GGAAGCAACTAAAACTACCGACGACAACGAAGAAGCACAATCTGAATTGCCCAAAATTGATGATTTGAAA\n",
      "ATCTCAGAGGCTGAAAAACCAAAAACTAAGGAGAATACCCCATCTGCTGATGATACTTCCTCAGAGAAGA\n",
      "CTACCAGCGCTAAGGCAGACACATCTACAGGAGGAGCGAACTCCGTGGATGCTCTAATCAAGGAACAAGA\n",
      "AGATGAGGTTGACGAAGAAGTCGTTAAAGATATGTTTGGTGGTAAGGATCATGTTTCCATCATTTTCATG\n",
      "GGTCACGTCGATGCTGGTAAGTCAACAATGGGTGGTAACTTGTTATATCTGACTGGTTCTGTGGATAAAA\n",
      "GAACCGTTGAGAAATATGAGAGAGAGGCTAAAGAGGCTGGTAGACAAGGTTGGTATTTATCATGGGTGAT\n",
      "GGATACCAACAAAGAAGAAAGAAACGACGGTAAAACCATTGAAGTGGGTAGAGCGTACTTTGAAACTGAA\n",
      "AAGAGACGTTACACTATTTTGGATGCTCCCGGTCACAAAATGTATGTTTCTGAAATGATTGGTGGTGCAT\n",
      "CTCAAGCCGATATTGGTATTTTGGTTATTTCTGCTAGAAAGGGTGAATATGAAACTGGTTTCGAAAAGGG\n",
      "TGGTCAAACTCGTGAGCACGCTTTATTAGCCAAGACACAAGGTGTCAACAAAATGATTGTAGTTATCAAC\n",
      "AAGATGGATGATCCAACTGTGGGATGGGATAAGGAAAGATATGATCACTGTGTTGGTAACTTGACAAACT\n",
      "TTTTGAAGGCTGTGGGTTACAATGTTAAGGAGGACGTCATTTTCATGCCAGTGTCTGGTTACACAGGTGC\n",
      "AGGTTTGAAGGAACGTGTCGATCCTAAGGACTGTCCATGGTATACTGGTCCATCTTTATTAGAATATCTT\n",
      "GACAATATGAAGACTACTGATCGTCATATCAATGCTCCATTCATGCTTCCAATTGCTTCTAAGATGAAGG\n",
      "ACATGGGTACTGTTGTGGAAGGTAAAATCGAATCTGGACACATTAGAAAGGGTAACCAAACTTTACTAAT\n",
      "GCCTAACAGGACCTCTGTTGAAATTCTGACCATTTATAACGAAACTGAAAGCGAAGTTGACATGGCTGTT\n",
      "TGTGGTGAGCAGGTTAGATTAAGAATTAAAGGTGTCGAAGAAGAAGAAATTTCTGCTGGTTTCGTTCTAA\n",
      "CCTCTCCAAAAAACCCAGTTAAGAATGTAACGAGATTTGTGGCTCAAATTGCTATTGTCGAATTGAAATC\n",
      "GATCATGTCTGCTGGTTTCTCGTGCGTTATGCATATTCATACAGCTATCGAAGAAGTCACCGTCACAAGA\n",
      "TTGCTTCACAAGCTTGAGAAGGGGTCAAACAGAAAATCAAAGAAGCCTCCAGCATTTGCTAAAAAGGGTA\n",
      "TGAAGATCATTGCCGTTATCGAGACTAATGAACCGGTATGTGTTGAAACATACGATGATTACCCACAATT\n",
      "GGGTAGATTCACTTTAAGAGATCAAGGTACCACCATTGCTATCGGTAAGATTGTGAAAATCCTTGAAAAT\n",
      "TGA\n",
      ">SUP35_Agos_ATCC_10895_NM_211584\n",
      "ATGTCGGAGGAAGATCAAATTCAATCGCAAGGCAACGACCAAGGCCAGTCGCAAGCCAAGGATCAAGGTC\n",
      "AAAATCAAGGTCAGGGGCAGCAAAATTTCGGCCAATACTACAACCCAAGTAACTTCCAGAATTACCAGGG\n",
      "GTATGTGCCTCAGGGAGGTTACCAGGCGTATGGCCAACAGGCAGGTGGGTACCAAGGCTATGCGCAATAC\n",
      "AACCAGCAAGCCGGCGGGTACCAGGGTTATCAGGGATACCAGCAGTACAACCCAGCACAAGCTGGCTACC\n",
      "AAGGTTACCAGCAATACAATGCGCAAGGTGGGTATCAGAGCTACAAGCAGTACAACTCACAGCCACAGGG\n",
      "GAACCGTAAGGGGAACCAGAGCTATGGTTACGGACAGGGCCAGTCAGCCACCGCTCCGGTGACGCTGAAC\n",
      "AACTTTGAGAAGGGAACTGTGCCGAATGCGACTGCTCCAAAACCAAAGAAGACCCTCAAGTTGGCTTCCA\n",
      "GCTCCGGTATCAAGCTAGTTGGTGCTAAGAAACCTGTCGCGAAGAAAGAAGAGGCTAAGGCGGAGGAGCC\n",
      "AACAAAGGAGGAGCCTAAGAGCAGCGCAGAGGGCGCTCCTAAGAGTGAGGATGCTACCGCTTCCTCTGAA\n",
      "GATAAGGCAGTCCCAAGTATTGAGAAGCTAAGTATTTCGGAAGCCGATACCGCGAAGAAAGACACAGCGG\n",
      "ACGCGGCCGGTGCCACCTCATCGGATGCTCTGATCAAGGAACAGGAAGACGAAGTTGATGAAGAAGTTGT\n",
      "GAAGGACATGTTTGGTGGAAAGGACCATGTTTCTATCATCTTTATGGGTCACGTCGATGCGGGGAAATCC\n",
      "ACCATGGGCGGAAACTTGTTGTATCTAACCGGCTCGGTAGACAAAAGAACAGTTGAGAAGTATGAGAAGG\n",
      "AGGCGAAGGAAGCAGGGAGACAGGGTTGGTACTTGTCTTGGATTATGGATACAAATAAGGAGGAGAGAAA\n",
      "CGATGGTAAGACCATTGAAGTGGGTAGATCCTACTTTGAAACTGAAAAGAGACGGTATACCATTTTGGAT\n",
      "GCACCAGGCCACAAAATGTACGTCTCTGAAATGATTGGTGGTGCGTCTCAAGCCGATGTTGGTATATTGG\n",
      "TCATCTCAGCCAGAAAGGGTGAGTACGAGACTGGTTTTGAGAGAGGTGGACAAACGCGTGAGCATGCGCT\n",
      "ATTAGCGAAGACCCAGGGTGTCAATAAAATGGTGGTCGTCGTGAACAAAATGGATGACCCCACGGTAAAC\n",
      "TGGGACAAGGCACGTTATGACCAGTGTATTAAGAACGTCTCCAATTTCTTGCAAGCTATTGGCTACAACG\n",
      "TTAAGGAGGATGTTATGTATATGCCTGTTTCTGGGTTTACTGGTGCTGGTCTGAAGGACCGCGTTGACAA\n",
      "GAAGGACTGTCCTTGGTACGATGGTCCTTCCCTTTTGGAATACCTAGACAACATGCAGCATGTTGACCGC\n",
      "TTCATCAACGCGCCTTTCATGCTTCCAATTGCAAGCAAGATGAAGGATATGGGTACAGTGGTCGAAGGTA\n",
      "AAATCGAATCTGGTCATATTAAGAAGGGTAATCAAACACTACTGATGCCAAACAAAATTCCAGTGGAGAT\n",
      "TCTAGCCATCCAAAATGAGACCGAACAGGAAGTTGATATGGCTGTGTGTGGTGAGCAGGTCAGATTGAGA\n",
      "CTGAAGGGAGTTGAGGAGGAAGACATTTCTGCAGGTTTTGTTTTGACTTCGCCAAAGAACCCTGTCAAGA\n",
      "ACGTCACGAAGTTTGTTGCCCAGATTGCCATTGTGGAATTGAAGTCCATTATGTCAGCAGGTTTCTCCTG\n",
      "TGTGATGCACGTGCATACCGCTATTGAGGAAGTCTCTATCACTAGACTACTTCATAAGCTGGAGAAGGGC\n",
      "ACCAACAGAAAGTCCAAGAAGCCACCTGCCTTTGCTAAGAAGGGCATGAAGATAATAGCGGTCCTTGAGA\n",
      "CAGAGGAGCCAGTATGTGTGGAAACCTACCAGGATTACCCACACTTGGGTAGATTCACATTGAGAGACCA\n",
      "GGGTATTACAATCGCCATCGGTAAGATTGTGAAGATCTTGGAATGA\n",
      ">SUP35_Scer_74-D694_GCA_001578265.1\n",
      "ATGTCGGATTCAAACCAAGGCAACAATCAGCAAAACTACCAGCAATACAGCCAGAACGGTAACCAACAAC\n",
      "AAGGTAACAACAGATACCAAGGTTATCAAGCTTACAATGCTCAAGCCCAACCTGCAGGTGGGTACTACCA\n",
      "AAATTACCAAGGTTATTCTGGGTACCAACAAGGTGGCTATCAACAGTACAATCCCGACGCCGGTTACCAG\n",
      "CAACAGTATAATCCTCAAGGAGGCTATCAACAGTACAATCCTCAAGGCGGTTATCAGCAGCAATTCAATC\n",
      "CACAAGGTGGCCGTGGAAATTACAAAAACTTCAACTACAATAACAATTTGCAAGGATATCAAGCTGGTTT\n",
      "CCAACCACAGTCTCAAGGTATGTCTTTGAACGACTTTCAAAAGCAACAAAAGCAGGCCGCTCCCAAACCA\n",
      "AAGAAGACTTTGAAGCTTGTCTCCAGTTCCGGTATCAAGTTGGCCAATGCTACCAAGAAGGTTGGCACAA\n",
      "AACCTGCCGAATCTGATAAGAAAGAGGAAGAGAAGTCTGCTGAAACCAAAGAACCAACTAAAGAGCCAAC\n",
      "AAAGGTCGAAGAACCAGTTAAAAAGGAGGAGAAACCAGTCCAGACTGAAGAAAAGACGGAGGAAAAATCG\n",
      "GAACTTCCAAAGGTAGAAGACCTTAAAATCTCTGAATCAACACATAATACCAACAATGCCAATGTTACCA\n",
      "GTGCTGATGCCTTGATCAAGGAACAGGAAGAAGAAGTGGATGACGAAGTTGTTAACGATATGTTTGGTGG\n",
      "TAAAGATCACGTTTCTTTAATTTTCATGGGTCATGTTGATGCCGGTAAATCTACTATGGGTGGTAATCTA\n",
      "CTATACTTGACTGGCTCTGTGGATAAGAGAACTATTGAGAAATATGAAAGAGAAGCCAAGGATGCAGGCA\n",
      "GACAAGGTTGGTACTTGTCATGGGTCATGGATACCAACAAAGAAGAAAGAAATGATGGTAAGACTATCGA\n",
      "AGTTGGTAAGGCCTACTTTGAAACTGAAAAAAGGCGTTATACCATATTGGATGCTCCTGGTCATAAAATG\n",
      "TACGTTTCCGAGATGATCGGTGGTGCTTCTCAAGCTGATGTTGGTGTTTTGGTCATTTCCGCCAGAAAGG\n",
      "GTGAGTACGAAACCGGTTTTGAGAGAGGTGGTCAAACTCGTGAACACGCCCTATTGGCCAAGACCCAAGG\n",
      "TGTTAATAAGATGGTTGTCGTCGTAAATAAGATGGATGACCCAACCGTTAACTGGTCTAAGGAACGTTAC\n",
      "GACCAATGTGTGAGTAATGTCAGCAATTTCTTGAGAGCAATTGGTTACAACATTAAGACAGACGTTGTAT\n",
      "TTATGCCAGTATCCGGCTACAGTGGTGCAAATTTGAAAGATCACGTAGATCCAAAAGAATGCCCATGGTA\n",
      "CACCGGCCCAACTCTGTTAGAATATCTGGATACAATGAACCACGTCGACCGTCACATCAATGCTCCATTC\n",
      "ATGTTGCCTATTGCCGCTAAGATGAAGGATCTAGGTACCATCGTTGAAGGTAAAATTGAATCCGGTCATA\n",
      "TCAAAAAGGGTCAATCCACCCTACTGATGCCTAACAAAACCGCTGTGGAAATTCAAAATATTTACAACGA\n",
      "AACTGAAAATGAAGTTGATATGGCTATGTGTGGTGAGCAAGTTAAACTAAGAATCAAAGGTGTTGAAGAA\n",
      "GAAGACATTTCACCAGGTTTTGTACTAACATCGCCAAAGAACCCTATCAAGAGTGTTACCAAGTTTGTAG\n",
      "CTCAAATTGCTATTGTAGAATTAAAATCTATCATAGCAGCCGGTTTTTCATGTGTTATGCATGTTCATAC\n",
      "AGCAATTGAAGAGGTACATATTGTTAAGTTATTGCACAAATTAGAAAAGGGTACCAACCGTAAGTCAAAG\n",
      "AAACCACCTGCTTTTGCTAAGAAGGGTATGAAGGTCATCGCTGTTTTAGAAACTGAAGCTCCAGTTTGTG\n",
      "TGGAAACTTACCAAGATTACCCTCAATTAGGTAGATTCACTTTGAGAGATCAAGGTACCACAATAGCAAT\n",
      "TGGTAAAATTGTTAAAATTGCCGAGTAA\n",
      ">SUP35_Spar_A12_Liti\n",
      "ATGTCGGATTCAAACCAAGGTAACAATCAGCAAAGCTACCAGCAATACGGCCAAAACCCTAACCAACAAC\n",
      "AAGGTAACAACAGATACCAAGGTTATCAAGCTTACAATGCTCAATCCCAACCTGCAGGTGGGTATTACCA\n",
      "AAATTACCAAGGTTATTCTGGATACCAACAAGGCAGCTACCAACAGTACAACCCAGATGCCGGTTACCAG\n",
      "CAACAATATAATCCCCAAGGTGGCTATCAACAGTACAATCCTCAAGGCGGTTATCAGCAACAGTTCAATC\n",
      "CACAAGGTGGCCGTGGCAATTACAAAAACTTCAACTACAATAACAATGCACAAGGATATCAAGCTGGTTT\n",
      "CCAACCACAGTCTCAAGGTATGTCTTTGAACGACTTCCAAAAGCAACAAAAGCAAGCCGCTCCCAAGCCA\n",
      "AAGAAAACTTTGAAGCTTGTTTCCAGTTCTGGTATCAAGTTGGCCAATGCTACCAAGAAGGTCGACACAA\n",
      "AACCTGCTGAATCTGAAAAGAAGAAGGAAGACAAACCCACCGAAAACAAAGAACCAACAAAACTCGAAGA\n",
      "ACCAGTTAAAAAAGAGGAGAAACTAGTCAAGACCGAAGAAAAAAAGGAGGAGAAATCGGAACTCCCAAAG\n",
      "GTAGAAGACCTGAAAATCTCTGAATCAACAGATAACACCAACAATGCCAATGTTAACAGTGCTGATGCCT\n",
      "TGATCAAAGAACAGGAAGAAGAAGTCGATGACGAAGTTGTTAACGACATGTTCGGGGGTAAAGATCACGT\n",
      "TTCCTTAATTTTCATGGGTCATGTTGATGCTGGTAAATCTACTATGGGTGGTAATCTACTGTACTTGACT\n",
      "GGCTCTGTCGATAAGAGAACCATTGAGAAATATGAAAGAGAGGCCAAGGATGCGGGAAGACAGGGTTGGT\n",
      "ACTTGTCATGGGTCATGGATACCAACAAAGAAGAAAGAAATGATGGTAAAACTATCGAAGTTGGTAAAGC\n",
      "CTACTTCGAAACTGAAAAAAGGCGTTATACCATATTGGATGCCCCTGGTCATAAAATGTACGTTTCCGAG\n",
      "ATGATCGGTGGTGCTTCTCAAGCTGATGTTGGTGTTTTGGTCATTTCCGCCAGAAAGGGTGAGTACGAAA\n",
      "CTGGTTTTGAAAGAGGTGGTCAAACACGTGAACACGCTCTATTAGCCAAGACCCAAGGTGTTAATAAGAT\n",
      "GGTTGTTGTCGTAAATAAGATGGATGACCCAACTGTTAACTGGTCTAAGGAACGTTATGACCAATGTGTG\n",
      "AGTAATGTCAGCAATTTCTTGAGAGCAATTGGCTACAATATCAAGACGGATGTTGTATTTATGCCAGTAT\n",
      "CTGGTTACAGTGGTGCAAACCTGAAAGATCACGTAGACCCAAAAGAATGCCCATGGTACACTGGCCCAAC\n",
      "TCTGTTGGAATATTTGGACACAATGAACCACGTCGACCGTCACATCAATGCTCCTTTCATGTTGCCTATT\n",
      "GCCGCTAAAATGAAGGATTTAGGTACCATCGTTGAAGGTAAGATTGAATCTGGTCATATCAAAAAGGGCC\n",
      "AATCCACGCTATTGATGCCTAACAAAACTGCAGTGGAAATTCAAAACATTTACAACGAAACAGAAAATGA\n",
      "AGTTGACATGGCTATGTGTGGTGAACAAGTTAAACTAAGAATCAAAGGTGTTGAAGAAGAAGACATTTCA\n",
      "CCAGGGTTCGTACTAACATCACCAAAGAACCCTATCAAAAGTGTTACCAAGTTTGTAGCTCAAATTGCAA\n",
      "TTGTCGAATTAAAATCTATTATTGCTGCTGGGTTTTCATGCGTTATGCATGTTCATACAGCAATTGAAGA\n",
      "AGTTCATATTGTTAAATTATTGCACAAATTGGAAAAGGGTACTAACCGTAAATCAAAGAAACCACCTGCG\n",
      "TTTGCTAAGAAAGGTATGAAGGTCATTGCTGTCTTAGAGACTGAAGCTCCAGTTTGTGTTGAAACTTACC\n",
      "AAGATTACCCTCAATTAGGTAGATTCACTTTGAGAGATCAAGGTACCACAATAGCAATTGGTAAGATTGT\n",
      "TAAAATTGCTGAATAA\n",
      ">SUP35_Smik_IFO1815T_30\n",
      "ATGTCTGATTCAAACCAAGGTAATAATCAGCAAAACTACCAGCAATACAACCAAAACCCCAACCAACAGC\n",
      "AAGGTAACAATAGATACCAAGGTTATCAAGCTTATAATGCTCAAGCGCAACCTGCAAGTGGCCATTACCA\n",
      "AAACTACCAAGGCTACTCTGGGTACCAACAAGGCGCTTTTCAACAGTACAACCCAGAGGCTGGCTACCAA\n",
      "CAACAGTACAACCCCCAAGGTGGCTATCAACAATACAATTCCCAGGGCGGTTATCAGCAACAATTCAATC\n",
      "CACAAGGCGGTCGTGGAAATTACCAAAACATCAATTATAACAGCAATATACAAGGATACCAAGCTGGTTT\n",
      "TCAACCACAGTCTCAAAGTATGTCTTTAAATGACTTCCAAAAGCAACAAAAGCAAGCTGCGCCTAAGCCA\n",
      "AAGAAGACTTTGAAGCTTGTTTCCAGTTCTGGTATCAAGCTAGCTAATGCTACCAAGAAAGTTGATACAA\n",
      "AACCTTCCGAAACTGAAAAGAAAGAGGAAGTCAAGCCTGTCGAAACCAAAAAACCAATTAAAGTTGAAGA\n",
      "ACCAGTCAAGAAAGAGGAAATGCCAGTAAAGGCCGAAGAAAAGAAGGAAGAAAAGTCTGAACTGCCAAAA\n",
      "GTAGGAGACTTAAAAATTTCTGAATCGACAGATAACAGCAACACTACCAGTGCTACCAGTGCTGATGCCT\n",
      "TGATCAAAGAACAGGAAGAAGAAGTTGATGATGAGGTTGTTAATGATATGTTCGGGGGCAAAGATCACGT\n",
      "TTCTTTAATCTTTATGGGTCATGTCGATGCTGGTAAGTCTACTATGGGTGGTAACTTACTGTACTTGACA\n",
      "GGCTCTGTGGATAAGAGAACTATAGAGAAATATGAAAGAGAGGCCAAGGATGCTGGTAGACAAGGTTGGT\n",
      "ATTTGTCATGGGTCATGGATACCAACAAAGAAGAAAGAAATGATGGTAAGACCATTGAAGTTGGTAAGGC\n",
      "CTACTTCGAAACTGAGAAAAGGCGTTATACTATATTGGATGCTCCTGGTCATAAAATGTACGTTTCCGAA\n",
      "ATGATTGGTGGTGCTTCTCAAGCTGATGTCGGTGTTTTAGTCATTTCTGCCAGAAAGGGTGAATACGAGA\n",
      "CTGGTTTTGAAAGAGGTGGTCAAACTCGTGAACATGCCTTATTGGCCAAGACCCAAGGTGTTAATAAGAT\n",
      "GGTTGTCGTCGTAAATAAAATGGATGATCCAACTGTCAATTGGTCCAAGGAACGTTATGATCAATGTGTA\n",
      "AGTAATGTCAGCAATTTCTTGAAAGCCATTGGCTATAACATCAAGACGGATGTTGTGTTTATGCCAGTAT\n",
      "CAGGTTACAGTGGTGCAAACTTGAAAAACCGTGTAGATCCAAAAGAATGTCCATGGTATACTGGCCCAAC\n",
      "TCTATTGGAGTATTTAGATACAATGAACCATGTCGATCGTCACATCAACGCTCCATTCATGTTGCCTATT\n",
      "GCCGCTAAGATGAAGGATCTGGGCACGATCGTTGAAGGTAAGATTGAATCTGGTCATATCAAGAAGGGTC\n",
      "AATCCACCCTATTAATGCCCAACAAAACGGCAGTGGAAATTCAAAACATTTATAATGAAACTGAAAACGA\n",
      "AGTTGACATGGCTATGTGTGGTGAGCAAGTCAAGCTGAGAATTAAGGGTGTTGAAGAAGAAGATATTTCC\n",
      "CCAGGGTTTGTACTAACATCACCAAAGAATCCTATCAAGAGTGTCACCAAGTTTGTGGCCCAAATTGCCA\n",
      "TTGTCGAACTAAAATCTATTATAGCTGCTGGGTTTTCATGTGTTATGCATGTTCATACAGCAATTGAAGA\n",
      "AGTTCATATTGTTAAATTATTGCACAAATTGGAAAAGGGTACTAACCGTAAATCAAAGAAGCCACCTGCT\n",
      "TTTGCCAAGAAGGGTATGAAAGTCATTGCTGTTTTAGAGACTGAAGCTCCGGTTTGTGTCGAAACTTACC\n",
      "AGGATTACCCTCAGTTAGGTAGATTCACCTTAAGAGATCAAGGTGCTACAATAGCAATTGGTAAGATCGT\n",
      "TAAGATTGCTGAATGA\n",
      ">SUP35_Skud_IFO1802T_36\n",
      "ATGTCAGATCCAAATCAAGGTAACAATCAACAACAATACGGTCAAAATCCTAACCAACAGCAAGGCAATA\n",
      "ACAAGTACCAAGGTTATCAAGCTTACAATGCTCAAGCCCAACCTGCAAACGGCTACTATCAAAACTACCA\n",
      "AGGATTTGCCGGCTATCAACAAGGCGGCTACCAACAGTATGACCGAGACGCCGGTAGTCAGCAACAATAC\n",
      "AACCCACAAGGTGGCTATCAACAGTACCCCCCACAGGAAGGTTACCAACAACAGTCCAATCCACAAGGTG\n",
      "GTCGTGGTAATTACAAAAGCTTCAACTATAACAACAATGTTCAAGGGTATCAAGCTGGCTTCCAACCGCA\n",
      "ATCTCAAGGTATGTCTCTGAACGACTTCCAAAAGCAACAAAAGCAAACTGCTCCTAAGCCAAAGAAGACT\n",
      "TTGAAGCTTGTTTCAAGCTCTGGTATCAAGTTAGCTAATGCTACGAAGAAGGTCAGCACAAAGCCTGCCG\n",
      "AAACCGAAAAGAAAGAGGAAGACAAGCCTGTCAAAACCGAGGAACAAATCAAGGTCGAAGAACCAATTAA\n",
      "AGAAGAGAAAAAAGCAGTCAAGACTGAGGAAAAGAAAGAGGTAGAATCTGAACTACCAAGGGTCGAAGAC\n",
      "TTGAAGATATCCGAATCGACTGATAATAGCAATACTGCTAACGTCACAAGTGCAGACACTTTGATAAAAG\n",
      "AACAAGAAGAAGAAGTTGATGACGAAGTTGTTAATGATATGTTTGGCGGTAAAGATCACGTTTCTTTGAT\n",
      "TTTCATGGGACATGTTGATGCTGGTAAGTCTACTATGGGTGGTAATCTGCTATACTTGACTGGCTCCGTG\n",
      "GATAAAAGAACAATTGAAAAATACGAAAGAGAAGCTAAGGATGCTGGTAGGCAAGGTTGGTACTTGTCAT\n",
      "GGGTCATGGATACAAATAAGGAAGAGAGAAACGACGGTAAGACCATTGAAGTTGGTAAAGCTTATTTCGA\n",
      "AACTGAGAAAAGACGTTATACCATTCTGGATGCTCCTGGTCATAAAATGTACGTTTCCGAAATGATTGGT\n",
      "GGTGCTTCGCAAGCCGATGTTGGTGTTTTAGTCATATCTGCCAGAAAGGGTGAATATGAAACCGGTTTCG\n",
      "AAAGAGGTGGCCAAACACGTGAACACGCCCTATTGGCCAAGACCCAGGGTGTTAATAAAATGGTTGTTGT\n",
      "CGTAAATAAGATGGATGATGCAACCGTCAACTGGTCCAAAGAACGTTACGATCAATGTGTAGGTAATGTT\n",
      "AGCAACTTTTTGAAAGCGATTGGTTACAACATCAAGACAGATGTTGCGTTTATGCCAGTATCAGGTTACA\n",
      "GCGGTGCAAACTTGAAAAATCGTGTAGATCCAAAGGAGTGTTCATGGTATACCGGCCCAACTCTGTTGGA\n",
      "ATACCTGGACACAATGAACCACGTCGATCGTCATGTCAATGCTCCATTCATGTTGCCTATTGCTGCTAAG\n",
      "ATGAAGGATTTAGGTACCATAGTTGAAGGCAAGATCGAATCCGGTCATATCAAGAAGGGTCAATCCACCT\n",
      "TGTTGATGCCTAACAAAACCTCAGTGGAAATTCAAAACATTTACAACGAAACTGAAAACGAAGTTGACAT\n",
      "GGCTATGTGCGGTGAACAGGTTAAGTTGAGAATTAAAGGAGTTGAAGAAGAAGACATCTCTCCAGGGTTT\n",
      "GTGTTAACATCACCAAAGAACCCTATCAAAAGTGTTACCAAATTTGTTGCCCAAATTGCCATTGTAGAGT\n",
      "TAAAGTCCATTATAGCTGCTGGGTTTTCATGTGTCATGCACGTTCATACAGCCATTGAAGAGGTTCATAT\n",
      "TGTTAAATTATTGCACAAATTAGAAAAGGGTAGCAATCGTAAATCGAAGAAGCCACCTGCTTTCGCTAAG\n",
      "AAAGGTATGAAGGTTATTGCTGTTTTAGAGACTGAAGCTCCAGTTTGTGTCGAAGCTTACCAAGATTATC\n",
      "CTCAATTGGGTAGATTCACTCTAAGAGATCAAGGTACCACAATCGCCATCGGTAAAATTGTTAAGATTGC\n",
      "TGAATAA\n",
      ">SUP35_Sbou_unique28_CM003560\n",
      "ATGTCGGATTCAAACCAAGGCAACAATCAGCAAAACTACCAGCAATACAGCCAGAACGGTAACCAACAAC\n",
      "AAGGTAACAACAGATACCAAGGTTATCAAGCTTACAATGCTCAAGCCCAACCTGCAGGTGGGTACTACCA\n",
      "AAATTACCAAGGTTATTCTGGGTACCAACAAGGTGGCTATCAACAGTACAATCCCGACGCCGGTTACCAG\n",
      "CAACAGTATAATCCTCAAGGAGGCTATCAACAGTACAATCCTCAAGGCGGTTATCAGCAGCAATTCAATC\n",
      "CACAAGGTGGCCGTGGAAATTACAAAAACTTCAACTACAATAACAATTTGCAAGGATATCAAGCTGGTTT\n",
      "CCAACCACAGTCTCAAGGTATGTCTTTGAACGACTTTCAAAAGCAACAAAAGCAGGCCGCTCCCAAACCA\n",
      "AAGAAGACTTTGAAGCTTGTCTCCAGTTCCGGTATCAAGTTGGCCAATGCTACCAAGAAGGTTGACACAA\n",
      "AACCTGCCGAATCTGATAAGAAAGAGGAAGAGAAGTCTGCTGAAACCAAAGAACCAACTAAAGAGCCAAC\n",
      "AAAGGTCGAAGAACCAGTTAAAAAGGAGGAGAAACCAGTCCAGACTGAAGAAAAGACGGAGGAAAAATCG\n",
      "GAACTTCCAAAGGTAGAAGACCTTAAAATCTCTGAATCAACACATAATACCAACAATGCCAATGTTACCA\n",
      "GTGCTGATGCCTTGATCAAGGAACAGGAAGAAGAAGTGGATGACGAAGTTGTTAACGATATGTTTGGTGG\n",
      "TAAAGATCACGTTTCTTTAATTTTCATGGGTCATGTTGATGCCGGTAAATCTACTATGGGTGGTAATCTA\n",
      "CTATACTTGACTGGCTCTGTGGATAAGAGAACTATTGAGAAATATGAAAGAGAAGCCAAGGATGCAGGCA\n",
      "GACAAGGTTGGTACTTGTCATGGGTCATGGATACCAACAAAGAAGAAAGAAATGATGGTAAGACTATCGA\n",
      "AGTTGGTAAGGCCTACTTTGAAACTGAAAAAAGGCGTTATACCATATTGGATGCTCCTGGTCATAAAATG\n",
      "TACGTTTCCGAGATGATCGGTGGTGCTTCTCAAGCTGATGTTGGTGTTTTGGTCATTTCCGCCAGAAAGG\n",
      "GTGAGTACGAAACCGGTTTTGAGAGAGGTGGTCAAACTCGTGAACACGCCCTATTGGCCAAGACCCAAGG\n",
      "TGTTAATAAGATGGTTGTCGTCGTAAATAAGATGGATGACCCAACCGTTAACTGGTCTAAGGAACGTTAC\n",
      "GACCAATGTGTGAGTAATGTCAGCAATTTCTTGAGAGCAATTGGTTACAACATTAAGACAGACGTTGTAT\n",
      "TTATGCCAGTATCCGGCTACAGTGGTGCAAATTTGAAAGATCACGTAGATCCAAAAGAATGCCCATGGTA\n",
      "CACCGGCCCAACTCTGTTAGAATATCTGGATACAATGAACCACGTCGACCGTCACATCAATGCTCCATTC\n",
      "ATGTTGCCTATTGCCGCTAAGATGAAGGATCTAGGTACCATCGTTGAAGGTAAAATTGAATCCGGTCATA\n",
      "TCAAAAAGGGTCAATCCACCCTACTGATGCCTAACAAAACCGCTGTGGAAATTCAAAATATTTACAACGA\n",
      "AACTGAAAATGAAGTTGATATGGCTATGTGTGGTGAGCAAGTTAAACTAAGAATCAAAGGTGTTGAAGAA\n",
      "GAAGACATTTCACCAGGTTTTGTACTAACATCGCCAAAGAACCCTATCAAGAGTGTTACCAAGTTTGTAG\n",
      "CTCAAATTGCTATTGTAGAATTAAAATCTATCATAGCAGCCGGTTTTTCATGTGTTATGCATGTTCATAC\n",
      "AGCAATTGAAGAGGTACATATTGTTAAGTTATTGCACAAATTAGAAAAGGGTACCAACCGTAAGTCAAAG\n",
      "AAACCACCTGCTTTTGCTAAGAAGGGTATGAAGGTCATCGCTGTTTTAGAAACTGAAGCTCCAGTTTGTG\n",
      "TGGAAACTTACCAAGATTACCCTCAATTAGGTAGATTCACTTTGAGAGATCAAGGTACCACAATAGCAAT\n",
      "TGGTAAAATTGTTAAAATCGCCGAGTAA\n",
      ">SUP35_Scer_beer078_CM005938\n",
      "ATGTCGGATTCAAACCAAGGCAACAATCAGCAAAACTA\n",
      "CCAGCAATACAGCCAGAACGGTAACCAACAACAAGGTAACAACAGATACCAAGGTTATCAAGCTTACAAT\n",
      "GCTCAAGCCCAACCTGCAGGTGGGTACTACCAAAATTACCAAGGTTATTCTGGGTACCAACAAGGTGGCT\n",
      "ATCAACAGTACAATCCTCAAGGCGGTTATCAGCAGCAATTCAATCCACAAGGTGGCCGTGGAAATTACAA\n",
      "AAACTTCAACTACAATAACAATTTGCAAGGATATCAAGCTGGTTTCCAACCACAGTCTCAAGGTATGTCT\n",
      "TTGAACGACTTTCAAAAGCAACAAAAGCAGGCCGCTCCCAAACCAAAGAAGACTTTGAAGCTTGTCTCCA\n",
      "GTTCCGGTATCAAGTTGGCCAATGCTACCAAGAAGGTTGACACAAAACCTGCCGAATCTGATAAGAAAGA\n",
      "GGAAGAGAAGTCTGCTGAAACCAAAGAACCAACTAAAGAGCCAACAAAGGTCGAAGAACCAGTTAAAAAG\n",
      "GAGGAGAAACCAGTCCAGACTGAAGAAAAGACGGAGGAAAAATCGGAACTTCCAAAGGTAGAAGACCTTA\n",
      "AAATCTCTGAATCAACACATAATACCAACAATGCCAATGTTACCAGTGCTGATGCCTTGATCAAGGAACA\n",
      "GGAAGAAGAAGTGGATGACGAAGTTGTTAACGATATGTTTGGTGGTAAAGATCACGTTTCTTTAATTTTC\n",
      "ATGGGTCATGTTGATGCCGGTAAATCTACTATGGGTGGTAATCTACTATACTTGACTGGCTCTGTGGATA\n",
      "AGAGAACTATTGAGAAATATGAAAGAGAAGCCAAGGATGCAGGCAGACAAGGTTGGTACTTGTCATGGGT\n",
      "CATGGATACCAACAAAGAAGAAAGAAATGATGGTAAGACTATCGAAGTTGGTAAGGCCTACTTTGAAACT\n",
      "GAAAAAAGGCGTTATACCATATTGGATGCTCCTGGTCATAAAATGTACGTTTCCGAGATGATCGGTGGTG\n",
      "CTTCTCAAGCTGATGTTGGTGTTTTGGTCATTTCCGCCAGAAAGGGTGAGTACGAAACCGGTTTTGAGAG\n",
      "AGGTGGTCAAACTCGTGAACACGCCCTATTGGCCAAGACCCAAGGTGTTAATAAGATGGTTGTCGTCGTA\n",
      "AATAAGATGGATGACCCAACCGTTAACTGGTCTAAGGAACGTTACGACCAATGTGTGAGTAATGTCAGCA\n",
      "ATTTCTTGAGAGCAATTGGTTACAACATTAAGACAGACGTTGTATTTATGCCAGTATCCGGCTACAGTGG\n",
      "TGCAAATTTGAAAGATCACGTAGATCCAAAAGAATGCCCATGGTACACCGGCCCAACTCTGTTAGAATAT\n",
      "CTGGATACAATGAACCACGTCGACCGTCACATCAATGCTCCATTCATGTTGCCTATTGCCGCTAAGATGA\n",
      "AGGATCTAGGTACCATCGTTGAAGGTAAAATTGAATCCGGTCATATCAAAAAGGGTCAATCCACCCTACT\n",
      "GATGCCTAACAAAACCGCTGTGGAAATTCAAAATATTTACAACGAAACTGAAAATGAAGTTGATATGGCT\n",
      "ATGTGTGGTGAGCAAGTTAAACTAAGAATCAAAGGTGTTGAAGAAGAAGACATTTCACCAGGTTTTGTAC\n",
      "TAACATCGCCAAAGAACCCTATCAAGAGTGTTACCAAGTTTGTAGCTCAAATTGCTATTGTAGAATTAAA\n",
      "ATCTATCATAGCAGCCGGTTTTTCATGTGTTATGCATGTTCATACAGCAATTGAAGAGGTACATATTGTT\n",
      "AAGTTATTGCACAAATTAGAAAAGGGTACCAACCGTAAGTCAAAGAAACCACCTGCTTTTGCTAAGAAGG\n",
      "GTATGAAGGTCATCGCTGTTTTAGAAACTGAAGCTCCAGTTTGTGTGGAAACTTACCAAGATTACCCTCA\n",
      "ATTAGGTAGATTCACTTTGAGAGATCAAGGTACCACAATAGCAATTGGTAAAATTGTTAAAATTGCCGAG\n",
      "TAA\n",
      ">SUP35_Sarb_H-6_chrXIII_CM001575\n",
      "ATGTCTGATCCAACTAATGGTAATAATGAGCAGAGCTCTCAACAGCAAGGCCAAAACCCTAGCCAACAGC\n",
      "AAGGTAACAACAGATATCAAGGCTATCAAGCTTATAATGCTCAAACTCAACCAGCAGGTGGCTATTACCA\n",
      "AAACTACCAAGGCTATGCTGGCTACCAACAAGGTGGATACCAACAGTTCACTCCAGAGGCTGGTTACCAA\n",
      "CAACAGTACAACCCCCAAGGTGGCTATCAACAGTTCAACCCCCAAGGCGGTTATCAACAACAGTTCAATC\n",
      "CACAAGGTGGCCGTGGCAACTACAAAAACTTTAACAATAACAACCAACAAGGATACCAAAGTAATTTTCA\n",
      "ACCGCAATCTCAAGGTATGTCTTTAAACGATTTCCAAAAGCAACAAAAGCAATCCACTCCTAAACCGAAG\n",
      "AAAACTTTGAAGCTTGTTTCAAGTTCCGGTATCAAGTTGGCTAATGCTACTAAGAAGGTCGATACAAAGC\n",
      "CCGTCGAAACCGAAAAGAAAGAAGAAGACAAGCCTGTTGAAACTAAAAAACCAACGAAGGTCGAAGAACC\n",
      "AGCTAAGAAAGAGGAAGAGCCCGTCAAGGCTGAAGAAGCAAAGGAGGAAAAATCAGAACTACCAAAGGTT\n",
      "GAAGACTTAAAAATATCTGAACCAACCGATAACAGCAACGCTGCTAGCGTCAACAATGCAGACGCCTTAA\n",
      "TTAAAGAACAAGAAGAAGAAGTCGATGATGAAGTTGTCAACGATATGTTTGGGGGCAAAGATCACGTTTC\n",
      "TTTGATCTTCATGGGTCATGTCGATGCTGGTAAGTCCACGATGGGTGGTAATCTGCTTTACTTGACCGGG\n",
      "TCCGTGGATAAGAGAACTATTGAAAAATATGAAAGAGAAGCTAAGGATGCCGGTAGACAAGGTTGGTATT\n",
      "TGTCATGGGTCATGGATACCAACAAAGAAGAAAGAAATGATGGTAAGACCATTGAAGTTGGTAAAGCTTA\n",
      "TTTCGAAACTGAAAAAAGACGTTATACCATTTTGGATGCTCCAGGTCATAAAATGTACGTTTCAGAAATG\n",
      "ATTGGTGGTGCCTCTCAAGCTGATGTTGGTGTCTTAGTAATTTCTGCCAGAAAGGGTGAATATGAAACTG\n",
      "GTTTCGAAAGAGGTGGTCAAACACGTGAACACGCTTTATTGGCCAAAACTCAAGGTGTTAATAAAATGGT\n",
      "TGTTGTCGTAAATAAGATGGACGACCCAACTGTCAACTGGTCCAAGGAACGTTACGACCAATGTGTAAGT\n",
      "AATGTCAGCAATTTCTTAAAGGCCATTGGTTACAATATCAAGACTGATGTTGTATTCATGCCAGTATCAG\n",
      "GTTACAGTGGTGCAAACTTGAAAGAACATGTAAATCCAAAAGAATGTTCATGGTACACCGGGCCAACTCT\n",
      "GTTGGAATACTTGGATAAAATGAATCACGTCGACCGTCACATCAATGCTCCATTCATGTTACCTATTGCT\n",
      "GCTAAGATGAAAGATCTAGGTACCATTGTAGAAGGTAAGATCGAATCCGGTCACATAAAGAAGGGTCAAT\n",
      "CAACTCTTTTAATGCCTAACAAAACATCAGTGGAAATTCAAAATATTTACAATGAAACTGAAAACGAAGT\n",
      "TGATATGGCTATGTGTGGTGAACAAGTTAAATTAAGAATTAAAGGTGTTGAAGAAGAAGATATTTCACCA\n",
      "GGGTTCGTCTTGACATCACCAAAGAACCCAATTAAGAGTGTTACCAAGTTTGTAGCTCAAATTGCCATTG\n",
      "TTGAACTGAAATCTATTATAGCTGCTGGGTTTTCATGTGTTATGCACGTTCACACAGCCATTGAAGAAGT\n",
      "TCATATCGTTAAGCTATTGCACAAGTTAGAAAAGGGTACTAACCGTAAATCGAAGAAGCCACCTGCTTTT\n",
      "GCTAAGAAGGGTATGAAGGTCATTGCTGTTCTAGAGACTGAAGCTCCAGTTTGTGTCGAGACTTACCAAG\n",
      "ACTACCCTCAATTGGGTAGATTCACTTTAAGAGACCAAGGTACCACAATCGCCATTGGTAAGATTGTTAA\n",
      "GATTGCTGAATAA\n",
      ">SUP35_Seub_CBS12357_chr_II_IV_DF968535\n",
      "ATGTCTGATCCAAACCAAGGTAACAATCAGCAAAACTATCAACAGTACGGTCAAAACTTCAACCAACAGC\n",
      "AAGGTAACAACAAATTTCAAGGTTACCAAGCTTACAATGCTCAAGCCCAACAACCTGCAGGTGGCTATTA\n",
      "CCAAAACCCCCAAGGTTACGCTGGCTACCAACAGGGCGGTTATGACCAACAATTTAACCCGGAAGTAGGT\n",
      "TACCAACAACAATACAACGCCCAAGGTGGTTACCAACAACAGTTCAATCCACAAGGTGGCCGTGGCAATT\n",
      "ACAAGAACTTCAACTACAACAACAGCCAACAGGGATTCCAAGCTGGCTTTCAACCACAATCTCAAGGAAT\n",
      "GTCTTTGAACGACTTCCAAAAGCAACAAAAACAAACTGCTCCTAAGCCAAAGAAGACTTTAAAACTTGTT\n",
      "TCAAGTTCCGGTATCAAGTTAGCTAATGCGACCAAGAAGGTCGACACAAAGCCTGTCGAAACTGAGAAGA\n",
      "AAGAGGAAGAAAAGCCTACCGAAACCAAAGAACCAGCCAAGGTCGAAGAATCAATCAAAGATGTGGAAAC\n",
      "TCCAGCCAGCGCTGAAGAAAAGAAGGAGGTGAATTCTGAATTACCAAAGGTCGAAGATTTGAAAATATCC\n",
      "GAATCGAACGATAACAGCAAGCCTGCTAACCCTATCAACGCCGATGCCTTGATCAAAGAACAAGAAGATG\n",
      "AAGTAGACGATGAAGTTGTTAATGATATGTTCGGAGGTAAAGATCACGTTTCTTTAATTTTCATGGGTCA\n",
      "CGTCGATGCTGGTAAGTCCACTATGGGTGGTAATCTACTATATTTGACTGGCTCTGTGGACAAGAGAACC\n",
      "ATTGAAAAATATGAAAGAGAAGCTAAAGATGCTGGTAGACAAGGTTGGTATTTGTCTTGGGTCATGGACA\n",
      "CCAATAAAGAAGAAAGAAACGACGGTAAGACCATTGAGGTCGGTAAGGCTTATTTCGAAACCGAAAAAAG\n",
      "ACGTTACACCATTCTGGATGCACCAGGTCATAAAATGTACGTTTCCGAAATGATTGGTGGTGCCTCTCAA\n",
      "GCCGATGTTGGTGTTCTAGTCATTTCTGCCAGAAAGGGTGAATACGAAACCGGTTTCGAAAGAGGTGGTC\n",
      "AAACACGTGAACACGCCTTATTGGCCAAGACCCAAGGTGTTAATAAGATGATCGTTGTCGTAAATAAGAT\n",
      "GGATGATCCTACTGTCAAATGGTCCAAAGAACGTTACGACCAATGTGTAGGTAACGTCAGCAATTTCTTA\n",
      "AAGGCCATCGGTTACAATATAAAGACTGATGTTATATTCATGCCAGTATCAGGTTACAGTGGTGCTAACT\n",
      "TGAAAGAACATGTAGATCCAAAGGAATGTTCATGGTACACCGGCCCAACTTTACTGGAATACTTGGACAA\n",
      "AATGACCCACGTTGACCGTCGTATCAACGCTCCATTCATGTTACCTATTGCTGCTAAAATGAAGGATTTA\n",
      "GGTACAATCGTAGAAGGTAAGATTGAATCTGGTCACATCAAGAAGGGCCAATCCACTTTATTGATGCCTA\n",
      "ATAAGACCACTGTGGAAATCCAAAACATCTATAACGAAACTGAAACCGAAGTCGATATGGCCATGTGTGG\n",
      "TGAACAAGTCAAACTAAGAATTAAAGGTGTCGAAGAAGACGACATTTCACCAGGGTTTGTGTTGACATCC\n",
      "CCAAAGAACCCAATTAAGAATGTTACCAAGTTTGTGGCGCAAATTGCCATTGTCGAATTAAAGTCCATTA\n",
      "TAGCCGCCGGATTTTCATGTGTTATGCATGTTCACACAGCCATTGAAGAGGTTCATATTGTTAAACTACT\n",
      "GCATAAGTTAGAAAAGGGTACAAATCGTAAATCGAAGAAGCCACCTGCTTTTGCTAAGAAGGGTATGAAG\n",
      "GTCATTGCTGTTTTAGAGACTGAAGCTCCAGTTTGCGTCGAGACTTACCAAGATTATCCACAGTTGGGTA\n",
      "GATTTACACTTAGAGACCAAGGTACCACAATTGCCATTGGTAAGATTGTCAAGATTGCTGAATGA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "seq = open('/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa').read()\n",
    "print(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#What sequences were used for analysis? Provide type of biopolymer, organism, gene name.\n",
    "\n",
    "#def find_organism(file):\n",
    "    \n",
    "    # get seqs from fasta file\n",
    "#    for record in SeqIO.parse(file,\"fasta\"):\n",
    "        # run BLAST\n",
    "#        blastResult = NCBIWWW.qblast(\"blastn\", \"nt\", record.seq)\n",
    "        # get first hit\n",
    "#        blastRecord = NCBIXML.read(blastResult)\n",
    "#        firstHit = blastRecord.alignments[0]\n",
    "        # get hit's gi number\n",
    "#        title = firstHit.title\n",
    "#        gi = title.split(\"|\")[1]\n",
    "        # search NCBI for the gi number\n",
    "#        ncbiResult = Entrez.efetch(db=\"nucleotide\", id=gi, rettype=\"gb\", retmode=\"text\")\n",
    " #       ncbiResultSeqRec = SeqIO.read(ncbiResult,\"gb\")\n",
    "#        first_organism = blastRecord.descriptions[0]\n",
    "\n",
    "#find_organism(\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1) What sequences were used for analysis? Provide type of biopolymer, organism, gene name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence number 0 is : >CP021242.1 Kluyveromyces lactis strain GG799 chromosome D, complete sequence\n",
      "\n",
      "Sequence number 1 is : >NM_211584.2 Eremothecium gossypii ATCC 10895 AGL145Wp (AGOS_AGL145W), partial \n",
      "\n",
      "Sequence number 2 is : >CP036483.1 Saccharomyces cerevisiae strain ySR128 chromosome IV, complete \n",
      "\n",
      "Sequence number 3 is : >CP020279.1 Saccharomyces paradoxus strain YPS138 chromosome IV, complete \n",
      "\n",
      "Sequence number 4 is : >LT986465.1 Saccharomyces jurei genome assembly, chromosome: IV\n",
      "\n",
      "Sequence number 5 is : >LR215954.1 Saccharomyces kudriavzevii strain CR85 genome assembly, chromosome: \n",
      "\n",
      "Sequence number 6 is : >CP020160.1 Saccharomyces cerevisiae strain DBVPG6765 chromosome IV, complete \n",
      "\n",
      "Sequence number 7 is : >CP004661.2 Saccharomyces cerevisiae YJM193 chromosome IV sequence\n",
      "\n",
      "Sequence number 8 is : >CP020279.1 Saccharomyces paradoxus strain YPS138 chromosome IV, complete \n",
      "\n",
      "Sequence number 9 is : >CP030946.1 Saccharomyces eubayanus strain CBS12357 chromosome II, complete \n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "fasta_record = SeqIO.parse(\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\", format='fasta')\n",
    "pattern = 'ALIGNMENTS\\n'\n",
    "for count, record in enumerate(fasta_record):\n",
    "    blastResult = list(NCBIWWW.qblast(\"blastn\", \"nt\", record.seq, format_type = 'text'))\n",
    "    start = blastResult.index(pattern)\n",
    "    origin = blastResult[start+1]\n",
    "    print(f'Sequence number {count} is : {origin}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2) Running different alignment algorithms (clustalw, muscle, clustalO, mafft and prank) for 10 DNA sequences (SUP35_10seqs.fa) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create comparison table with running time\n",
    "\n",
    "from time import process_time "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.002311546999999914 seconds\n"
     ]
    }
   ],
   "source": [
    "#sudo apt-get install clustalw\n",
    "\n",
    "\n",
    "import Bio.Align.Applications\n",
    "from Bio.Align.Applications import ClustalwCommandline\n",
    "\n",
    "clustalw = r\"/usr/bin/clustalw\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\"\n",
    "out_file = r\"./clustalw/Clustal10.fasta\"\n",
    "start = process_time()\n",
    "cline_clustalw = ClustalwCommandline(\"clustalw\", infile = in_file, outfile = out_file  )\n",
    "stop = process_time() \n",
    "t1 = stop - start\n",
    "print(t1, \"seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0010732939999997804 seconds\n"
     ]
    }
   ],
   "source": [
    "# Muscle alignment\n",
    "\n",
    "from Bio.Align.Applications import MuscleCommandline\n",
    "\n",
    "muscle = r\"/usr/bin/muscle\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\"\n",
    "out_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/process_alignments/muscle/Muscle10.fasta\"\n",
    "start = process_time()\n",
    "cline_Muscle = MuscleCommandline(muscle, input = in_file, out = out_file)\n",
    "stop = process_time() \n",
    "t2 = stop - start\n",
    "print(t2, \"seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.011364158000000124 seconds\n"
     ]
    }
   ],
   "source": [
    "# Prank alignment\n",
    "#sudo apt-get update -y\n",
    "#sudo apt-get install -y prank\n",
    "\n",
    "from Bio.Align.Applications import PrankCommandline\n",
    "import os\n",
    "\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\"\n",
    "start = process_time()\n",
    "prank_cline = PrankCommandline(d = in_file,\n",
    "                               o =\"aligned\",\n",
    "                               f = 8, # FASTA output\n",
    "                               notree = True, noxml = True)\n",
    "prank_cline()\n",
    "stop = process_time() \n",
    "t3 = stop - start\n",
    "print(t3, \"seconds\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.014952075999999925 seconds\n"
     ]
    }
   ],
   "source": [
    "# Mafft alignment:\n",
    "\n",
    "#sudo apt install mafft\n",
    "\n",
    "from Bio.Align.Applications import MafftCommandline\n",
    "\n",
    "mafft = r\"/usr/bin/mafft\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\"\n",
    "start = process_time()\n",
    "mafft_cline = MafftCommandline(mafft, input=in_file) \n",
    "stdout, stderr = mafft_cline() \n",
    "with open(\"mafft10.fasta\", \"w\") as handle: \n",
    "    handle.write(stdout)\n",
    "mafft_cline()\n",
    "stop = process_time() \n",
    "t4 = stop - start\n",
    "print(t4, \"seconds\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0008695830000000626 seconds\n"
     ]
    }
   ],
   "source": [
    "# ClustalOmega alignment:\n",
    "#sudo apt-get install clustalo\n",
    "\n",
    "from Bio.Align.Applications import ClustalOmegaCommandline\n",
    "\n",
    "clustalo = r\"/usr/bin/clustalo\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs.fa\"\n",
    "out_file = r\"./clustalo/ClustalOmega.fasta\"\n",
    "start = process_time()\n",
    "cline_ClustalO = ClustalOmegaCommandline(clustalo, infile=in_file, outfile=out_file, verbose=True, auto=True)\n",
    "stop = process_time() \n",
    "t5 = stop - start\n",
    "print(t5, \"seconds\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### running time comparison table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+-----------+-----------------------+\n",
      "| algorithm |      Running_time     |\n",
      "+-----------+-----------------------+\n",
      "|  ClustalW |  0.002311546999999914 |\n",
      "|   Muscle  | 0.0010732939999997804 |\n",
      "|   Prank   |  0.011364158000000124 |\n",
      "|   Mafft   |  0.014952075999999925 |\n",
      "|  ClustalO | 0.0008695830000000626 |\n",
      "+-----------+-----------------------+\n"
     ]
    }
   ],
   "source": [
    "from prettytable import PrettyTable\n",
    "x = PrettyTable()\n",
    "x.field_names = ['algorithm', 'Running_time']\n",
    "x.add_row(['ClustalW', t1])\n",
    "x.add_row(['Muscle', t2])\n",
    "x.add_row(['Prank', t3])\n",
    "x.add_row(['Mafft', t4])\n",
    "x.add_row(['ClustalO', t5])\n",
    "print(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on running time I can tell that clustalo if faster that other algorithms!\n",
    "\n",
    "the question that arises here is I have read muscle is more accurate than T-Coffee and faster than Clustal-W! but why results does not show this!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### comments on the DNA alignment quality for the 5 algorithms. Which algorithm is better to use?\n",
    "\n",
    "\n",
    "Clustalo shows better results I think because it contain less gaps compared with other methods.\n",
    "\n",
    "But I read this text about alignments:\n",
    "\n",
    "Muscle has a better theoretical basis than Clustal. Clustal is highly sensitive to the order in which you list the sequences, because its search algorithm always depends on the order, especially the first sequence.\n",
    "Muscle takes an iterative approach, which is a plus. The advantage of an iterative approach is that it can search through a variety of alignments and phylogenies, make changes to them, test if it's an improvement or not, and continue onward until it has a more optimal solution."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3) What is wrong with the alignment of SUP35_10seqs_strange_aln.fa and how to fix it?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Converting a file of sequences to their reverse complements\n",
    "\n",
    "\n",
    "from Bio import SeqIO\n",
    "sequences = (seq.reverse_complement(id=\"rc_\"+seq.id, description = \"reverse complement\") \\\n",
    "           for seq in SeqIO.parse(\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/homework_3/SUP35_10seqs_strange_aln.fa\", \"fasta\"))\n",
    "SeqIO.write(sequences, \"rev_comp.fasta\", \"fasta\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4) Obtain amino acid sequences for these data. Repeat p.2 on amino acid sequences."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# To get amino acids we need to Tranlate DNA to protein:\n",
    "https://github.com/prestevez/dna2proteins\n",
    "\n",
    "I used python code and got protein sequence from this command in terminal:\n",
    "\n",
    "sample command:\n",
    "python dna2proteins.py -i sequences.fa -o proteins.fa -p\n",
    "\n",
    "    \n",
    "python /home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/dna2proteins.py -i /home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/SUP35_10seqs.fa  -o /home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#second approach\n",
    "\n",
    "file = open('/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/SUP35_10seqs.fa', 'r')\n",
    "dna = file.read()\n",
    "\n",
    "#print (\"DNA Sequence: \", dna)\n",
    "\n",
    "# DNA codon table\n",
    "protein = {\"TTT\" : \"F\", \"CTT\" : \"L\", \"ATT\" : \"I\", \"GTT\" : \"V\",\n",
    "           \"TTC\" : \"F\", \"CTC\" : \"L\", \"ATC\" : \"I\", \"GTC\" : \"V\",\n",
    "           \"TTA\" : \"L\", \"CTA\" : \"L\", \"ATA\" : \"I\", \"GTA\" : \"V\",\n",
    "           \"TTG\" : \"L\", \"CTG\" : \"L\", \"ATG\" : \"M\", \"GTG\" : \"V\",\n",
    "           \"TCT\" : \"S\", \"CCT\" : \"P\", \"ACT\" : \"T\", \"GCT\" : \"A\",\n",
    "           \"TCC\" : \"S\", \"CCC\" : \"P\", \"ACC\" : \"T\", \"GCC\" : \"A\",\n",
    "           \"TCA\" : \"S\", \"CCA\" : \"P\", \"ACA\" : \"T\", \"GCA\" : \"A\",\n",
    "           \"TCG\" : \"S\", \"CCG\" : \"P\", \"ACG\" : \"T\", \"GCG\" : \"A\",\n",
    "           \"TAT\" : \"Y\", \"CAT\" : \"H\", \"AAT\" : \"N\", \"GAT\" : \"D\",\n",
    "           \"TAC\" : \"Y\", \"CAC\" : \"H\", \"AAC\" : \"N\", \"GAC\" : \"D\",\n",
    "           \"TAA\" : \"STOP\", \"CAA\" : \"Q\", \"AAA\" : \"K\", \"GAA\" : \"E\",\n",
    "           \"TAG\" : \"STOP\", \"CAG\" : \"Q\", \"AAG\" : \"K\", \"GAG\" : \"E\",\n",
    "           \"TGT\" : \"C\", \"CGT\" : \"R\", \"AGT\" : \"S\", \"GGT\" : \"G\",\n",
    "           \"TGC\" : \"C\", \"CGC\" : \"R\", \"AGC\" : \"S\", \"GGC\" : \"G\",\n",
    "           \"TGA\" : \"STOP\", \"CGA\" : \"R\", \"AGA\" : \"R\", \"GGA\" : \"G\",\n",
    "           \"TGG\" : \"W\", \"CGG\" : \"R\", \"AGG\" : \"R\", \"GGG\" : \"G\" \n",
    "           }\n",
    "\n",
    "protein_sequence = \"\"\n",
    "\n",
    "# Generate protein sequence\n",
    "for i in range(0, len(dna)-(3+len(dna)%3), 3):\n",
    "    if protein[dna[i:i+3]] == \"STOP\" :\n",
    "        break\n",
    "    protein_sequence += protein[dna[i:i+3]]\n",
    "\n",
    "# Print the protein sequence\n",
    "print (\"Protein_Sequence: \", protein_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.01476589099999992 seconds\n"
     ]
    }
   ],
   "source": [
    "# Muscle:\n",
    "\n",
    "from Bio.Align.Applications import MuscleCommandline \n",
    "muscle_exe = r\"/usr/bin/muscle\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\"\n",
    "out_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/Muscle10p.fasta\"\n",
    "\n",
    "start = process_time()\n",
    "cline = MuscleCommandline(muscle_exe, input=in_file, out =out_file)\n",
    "cl = str(cline)\n",
    "stdout, stderr = cline()\n",
    "stop = process_time() \n",
    "t_p1 = stop - start\n",
    "print(t_p1, \"seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0029270630000000075 seconds\n"
     ]
    }
   ],
   "source": [
    "# ClustalW:\n",
    "import Bio.Align.Applications\n",
    "from Bio.Align.Applications import ClustalwCommandline\n",
    "\n",
    "clustalw = r\"/usr/bin/clustalw\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\"\n",
    "out_file = r\"./clustalw/Clustal10p.fasta\"\n",
    "start = process_time()\n",
    "cline_clustalw = ClustalwCommandline(\"clustalw\", infile = in_file, outfile = out_file  )\n",
    "stop = process_time() \n",
    "t_p2 = stop - start\n",
    "print(t_p2, \"seconds\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.007622177000000008 seconds\n"
     ]
    }
   ],
   "source": [
    "# Prank:\n",
    "\n",
    "\n",
    "from Bio.Align.Applications import PrankCommandline\n",
    "import os\n",
    "\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\"\n",
    "start = process_time()\n",
    "prank_cline = PrankCommandline(d = in_file,\n",
    "                               o =\"aligned\",\n",
    "                               f = 8, # FASTA output\n",
    "                               notree = True, noxml = True)\n",
    "prank_cline()\n",
    "stop = process_time() \n",
    "t_p3 = stop - start\n",
    "print(t_p3, \"seconds\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.012345296000000117 seconds\n"
     ]
    }
   ],
   "source": [
    "# mafft:\n",
    "\n",
    "\n",
    "from Bio.Align.Applications import MafftCommandline\n",
    "\n",
    "mafft_exe = r\"/usr/bin/mafft\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\"\n",
    "\n",
    "start = process_time()\n",
    "mafft_cline = MafftCommandline(mafft_exe, input=in_file) \n",
    "stdout, stderr = mafft_cline() \n",
    "with open(\"mafft10p.fasta\", \"w\") as handle: \n",
    "    handle.write(stdout)\n",
    "mafft_cline()\n",
    "stop = process_time() \n",
    "t_p4 = stop - start\n",
    "print(t_p4, \"seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0071628729999999585 seconds\n"
     ]
    }
   ],
   "source": [
    "# ClustalOmega:\n",
    "\n",
    "from Bio.Align.Applications import ClustalOmegaCommandline\n",
    "clustalo_exe = r\"/usr/bin/clustalo\"\n",
    "in_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/protein.fasta\"\n",
    "out_file = r\"/home/sedreh/ITMO/semester3/Molecular_phylogenetic/phylogenetics_part1/homework_3/Biopython_process/protein/ClustalO10p.fasta\"\n",
    "\n",
    "start = process_time()\n",
    "cline = ClustalOmegaCommandline(clustalo_exe, infile=in_file, outfile=out_file, force = True)\n",
    "stdout, stderr = cline()\n",
    "cline()\n",
    "stop = process_time() \n",
    "t_p5 = stop - start\n",
    "print(t_p5, \"seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+-----------+-----------------------+\n",
      "| algorithm |      Running_time     |\n",
      "+-----------+-----------------------+\n",
      "|  ClustalW | 0.0029270630000000075 |\n",
      "|   Muscle  |  0.01476589099999992  |\n",
      "|   Prank   |  0.007622177000000008 |\n",
      "|   Mafft   |  0.012345296000000117 |\n",
      "|  ClustalO | 0.0071628729999999585 |\n",
      "+-----------+-----------------------+\n"
     ]
    }
   ],
   "source": [
    "# Comparison table of algorithms running time for protein:\n",
    "\n",
    "from prettytable import PrettyTable\n",
    "x = PrettyTable()\n",
    "x.field_names = ['algorithm', 'Running_time']\n",
    "x.add_row(['ClustalW', t_p2])\n",
    "x.add_row(['Muscle', t_p1])\n",
    "x.add_row(['Prank', t_p3])\n",
    "x.add_row(['Mafft', t_p4])\n",
    "x.add_row(['ClustalO', t_p5])\n",
    "print(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5) Optional Repeat p. 2 on the alignment of 250 DNA sequences (SUP35_250seqs.fa). Has our choice of algorithm changed?\n",
    "\n",
    "\n",
    "I have done it and attached files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6) How to add to the alignment another sequence (SUP35_1addseq.fsa) using mafft and muscle?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#adding new sequence to the alignment using mucsle\n",
    "#http://www.drive5.com/muscle/muscle_userguide3.8.html\n",
    "\n",
    "#To add a sequence to an existing alignment, use profile-profile alignment with the new sequence as a profile.\n",
    "For example, if you have an existing alignment Muscle10.fasta and want to add a new sequence in SUP35_1addseq.fsa, use the following commands:\n",
    "muscle -profile -in1 existing_aln.afa -in2 new_seq.fa -out combined.afa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#adding new sequence to the alignment using mafft\n",
    "#https://mafft.cbrc.jp/alignment/software/addsequences.html\n",
    "\n",
    "% mafft --add new_sequences --reorder existing_alignment > output\n",
    "\n",
    "\n",
    "% mafft --addfragments fragments --reorder --thread -1 existing_alignment > output\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### # 7) How to add two more sequences (SUP35_2addseqs.fsa), pre-aligning them with the mafft or muscle?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8) #Try to run Gblocks (for alignment of 10 and 250 sequences). What percent of alignment remains after starting Gblocks with strict and non-strict parameters(specify parameters)?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://molevol.cmima.csic.es/castresana/Gblocks_server.html\n",
    "\n",
    "            ######## strict parameters #######\n",
    "            \n",
    "                ######  86% of the original 2153 positions#####\n",
    "\n",
    "Parameters used\n",
    "Minimum Number Of Sequences For A Conserved Position: 6\n",
    "Minimum Number Of Sequences For A Flanking Position: 8\n",
    "Maximum Number Of Contiguous Nonconserved Positions: 4\n",
    "Minimum Length Of A Block: 10\n",
    "Allowed Gap Positions: None\n",
    "\n",
    "Flank positions of the 15 selected block(s)\n",
    "Flanks: [33  44]  [67  84]  [118  172]  [205  224]  [232  251]  [311  331]  [338  371]  [399  421]  [425  470]  [474  487]  [492  610]  [654  697]  [708  720]  [722  740]  [743  2150]  \n",
    "\n",
    "New number of positions in ClustalOmega.fasta-gb:  1866  (86% of the original 2153 positions)\n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "           \n",
    "          ######## non-strict parameters ########\n",
    "          \n",
    "              ###### 94% of the original 2153 positions #######\n",
    "\n",
    "Parameters used\n",
    "Minimum Number Of Sequences For A Conserved Position: 6\n",
    "Minimum Number Of Sequences For A Flanking Position: 6\n",
    "Maximum Number Of Contiguous Nonconserved Positions: 8\n",
    "Minimum Length Of A Block: 5\n",
    "Allowed Gap Positions: With Half\n",
    "\n",
    "Flank positions of the 16 selected block(s)\n",
    "Flanks: [1  8]  [33  62]  [67  90]  [118  178]  [182  190]  [194  224]  [231  331]  [338  371]  [377  470]  [474  487]  [492  610]  [614  631]  [644  697]  [701  720]  [722  740]  [743  2150]  \n",
    "\n",
    "New number of positions in ClustalOmega.fasta-gb:  2044  (94% of the original 2153 positions)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
